controllers/controllers/resource/reconciler_test.go

```diff
@@ -152,13 +152,23 @@ func TestClusterReconcilerReconcile(t *testing.T) {
 				if err := yaml.Unmarshal([]byte(machineDeploymentFile), machineDeployment); err != nil {
 					t.Errorf("unmarshalling machinedeployment failed: %v", err)
 				}
+				workerNodeMachineConfig := &anywherev1.VSphereMachineConfig{
```

Comment:
this pr also covers controller updates? if not fully, let's maybe move the controller changes into separate pr

Reply:
There is common code that calls the adding new kubeadm config template, which is why these changes were necessary. However the immutable changes for ssh keys in the webhook haven't been updated, so the updates to the ssh keys wont go through.